convert/jenkinsjson/convert.go

```diff
@@ -963,6 +965,10 @@ func canMergeSteps(step1, step2 *harness.Step) bool {
 		return false
 	}
 
+	if !hasDefaultOrNoImage(exec1) || !hasDefaultOrNoImage(exec2) {
```

Comment:
In the case where both step1 and step2 have no image, I assume they could be merged since the image matches?

Reply:
The hasDefaultOrNoImage check ensures that steps can only be merged if they both use either no image or the default image. Any custom images, even if they're from the same base image but different versions, will not be merged. For example:

1. golang:latest and golang:1.22.4 won't merge
2. alpine and alpine:14 won't merge
3. Empty image and empty image will merge
4. Default image and empty image will merge
This is important because different image versions could have different tools, dependencies, or behaviors, which could cause unexpected results if merged. The current code properly handles this by preventing merging of steps with any custom images.
This is just a safer approach, in my opinion, and I also tested it, and it is working as expected.
I have tested it with most of the steps whose conversion code is a shell script step.